jlm/llvm/opt/alias-analyses/Andersen.cpp

```diff
@@ -150,77 +141,7 @@ class Andersen::Statistics final : public jlm::util::Statistics
         "AndersenAnalysis ",
         SourceFile_.to_str(),
         " ",
-        "#RvsdgNodes:",
-        NumRvsdgNodes_,
-        " ",
-        "AliasAnalysisTime[ns]:",
-        AnalysisTimer_.ns(),
-        " ",
-        "#PointerObjects:",
-        NumPointerObjects_,
-        " ",
-        "#RegistersMappedToPointerObject:",
-        NumRegistersMappedToPointerObject_,
-        " ",
-        "#SupersetConstraints:",
-        NumSupersetConstraints_,
-        " ",
-        "#NumAllPointeesPointToSupersetConstraints:",
-        NumAllPointeesPointToSupersetConstraints_,
-        " ",
-        "#SupersetOfAllPointeesConstraints:",
-        NumSupersetOfAllPointeesConstraints_,
-        " ",
-        "#HandleEscapingFunctionConstraints:",
-        NumHandleEscapingFunctionConstraints_,
-        " ",
-        "#FunctionCallConstraints:",
-        NumFunctionCallConstraints_,
-        " ",
-        "#ConstraintSolvingIterations:",
-        NumConstraintSolvingIterations_,
-        " ",
-        "#PointsToGraphNodes:",
-        NumPointsToGraphNodes_,
-        " ",
-        "#RegisterNodes:",
-        NumRegisterNodes_,
-        " ",
-        "#AllocaNodes:",
-        NumAllocaNodes_,
-        " ",
-        "#DeltaNodes:",
-        NumDeltaNodes_,
-        " ",
-        "#ImportNodes:",
-        NumImportNodes_,
-        " ",
-        "#LambdaNodes:",
-        NumLambdaNodes_,
-        " ",
-        "#MallocNodes:",
-        NumMallocNodes_,
-        " ",
-        "#MemoryNodes:",
-        NumMemoryNodes_,
-        " ",
-        "#EscapedNodes:",
-        NumEscapedNodes_,
-        " ",
-        "#ExternalMemorySources:",
-        NumExternalMemorySources_,
-        " ",
-        "SetAndConstraintBuildingTime[ns]:",
-        SetAndConstraintBuildingTimer_.ns(),
-        " ",
-        "ConstraintSolvingTime[ns]:",
-        ConstraintSolvingTimer_.ns(),
-        " ",
-        "PointsToGraphConstructionTime[ns]:",
-        PointsToGraphConstructionTimer_.ns(),
-        " ",
-        "PointsToGraphConstructionExternalToEscapedTime[ns]:",
-        PointsToGraphConstructionExternalToEscapedTimer_.ns());
+        SerializeMeasurements());
```

Comment:
Could you add another `SerializeMeasurements()` method that prefixes the measurements with the Statistics::Id followed by the source file. Ultimately, this is what we want (and after we changed everything to use that one, we remove the ToString() method). However, I understand that the current method is necessary in order to not have the changes all at once all over the place. I can take care in the end to refactor all the other statistics.

Reply:
I tried doing something like that at first (giving `ToString` a default implementation) but encountered two issues:

 - where can I find the ID as a string? I'm guessing it eventually should correspond to the ID used on the command line, but right now there is no source of truth location for translating between the enum and a string.
 - not all Statistics have a field for the file name. The `RegionAwareMemoryNodeProvider`'s statistics don't include it in it's output. Should all Statistics have the fileName included? If so, we can maybe move the field to the superclass and add it to the ctor?

Reply:
1. As far as I know, there is currently no method for providing a string representation of a statistics ID. C++ does not come equipped with it, and I have not implemented one so far. You would need to implement it. We have methods for translating the statistics IDs to command line flags though. See [here](https://github.com/phate/jlm/blob/94ba20fd93ace0da2a0b52df6891cc3b46051bb4/jlm/tooling/CommandLine.cpp#L180C1-L269C2). I am a little bit reluctant of tying the internal string representation of an statistics id to the command line flag though. It would mean that by changing the name of an id (and its string representation), one would all in a sudden also change the command line flag, which is a public interface. I think they should be separate concerns.
2. Yes, all statistics should have the filename included. It makes no sense to leave this (important) piece of information out. Moving it to the superclass and adding it to the ctor would be the right thing to do.